Src/baro_proc.c

```diff
@@ -0,0 +1,70 @@
+#include "main.h"
+#include "baro_LPS22HB.h"
+
+#define BARO_AVG_COUNT 4
+
+static uint32_t baro_value[BARO_AVG_COUNT];
+static uint32_t baro_temp_value[BARO_AVG_COUNT];
+static int32_t baro_value_idx;
+
+/**
+  * @brief Initialize baro (LPS22HB)
+  * @retval None
+  */
+void baroAvgInit()
+{
+	int16_t init_temp_value;
+	uint32_t init_value= baroReadPressTemp(&init_temp_value);
+
+	// set all buf to the inital value
+	for(int i=0;i<BARO_AVG_COUNT;i++)
+	{
+		baro_value[i]=init_value;
+		baro_temp_value[i]=init_temp_value;
+	}
+
+	baro_value_idx=0;
+}
+
+void baroSample()
+{
+	int16_t baro_temp_reading;
+	uint32_t baro_reading= baroReadPressTemp(&baro_temp_reading);
+
+	if(baro_reading!=0xffffffff)
+	{// save the value if it is valid
+		baro_value[baro_value_idx]=baro_reading;
+		baro_temp_value[baro_value_idx]=baro_temp_reading;
+		baro_value_idx++;
+
+		if(baro_value_idx>=BARO_AVG_COUNT)
+		{
+			baro_value_idx=0;
+		}
+	}
+
+}
+
+uint32_t baroGetAvg()
+{
+	uint32_t total=0;
+
+	for(int i=0;i<BARO_AVG_COUNT;i++)
+	{
+		total+=baro_value[i];
+	}
+
+	return total/BARO_AVG_COUNT;
+}
+
+int16_t baroGetTempAvg()
+{
+	int32_t total=0;
+
+	for(int i=0;i<BARO_AVG_COUNT;i++)
+	{
+		total+=baro_temp_value[i];
+	}
+
+	return total/BARO_AVG_COUNT;
+}

```

Comment:
Ideally, the average calculations should be happening before the controller requests new data (such as every time a new sample is acquired). That way, when the controller requests data, there are less processes running in the ISR where the packet is being built, and the response goes out quicker.
In addition to that, it would be nice if this is changed to use an EMA function rather than an array. It will be much easier to adjust the averaging that way as you mess with data rate settings, rather than messing with the average sample array size.